litho-processor/src/main/java/com/facebook/litho/specmodels/model/DefaultLayoutSpecGenerator.java

```diff
@@ -46,6 +47,7 @@ public TypeSpec generate(LayoutSpecModel layoutSpecModel, EnumSet<RunMode> runMo
     final TypeSpec.Builder typeSpec =
         TypeSpec.classBuilder(layoutSpecModel.getComponentName())
             .superclass(ClassNames.COMPONENT)
+            .addOriginatingElement((TypeElement) layoutSpecModel.getRepresentedObject())
```

Comment:
Is this actually being used somewhere now? And if not, are we still getting the speed-up benefits from what it is now?

Reply:
The following is still true, despite reverting the changes to the `SpecModel`:
> Add originating element when creating a `*SpecModel`.
> * This is required, in order to pass it on the `TypeSpec` that in turn forwards those to `Filer API`.

Reply:
Where is the code with the Filer API? I'm not seeing much code remaining after the latest update :)

Reply:
You can check this test from Javapoet, which does exactly this assertion https://github.com/square/javapoet/blob/master/src/test/java/com/squareup/javapoet/FileWritingTest.java 

Also please check `Filer` docs: https://docs.oracle.com/javase/7/docs/api/javax/annotation/processing/Filer.html 

> If there are no originating elements, none need to be passed. This information may be used in an incremental environment to determine the need to rerun processors or remove generated files. Non-incremental environments may ignore the originating element information.

Reply:
Oh my mistake, I read this totally wrong! Thanks for your patience :) I leave this to @passy

Reply:
No problem! Happy to explain!

Reply:
@IanChilds raised to me that there is a possibility of the represented object being a `PsiClass`. Could you add a check here before casting to ensure that we don't blow up in that case? And sorry for the churn on this.